apps/avatax/src/modules/saleor/order-confirmed/event.ts

```diff
@@ -0,0 +1,102 @@
+import { Result, err, ok } from "neverthrow";
+import { z } from "zod";
+import { BaseError } from "../../../error";
+import { OrderConfirmedPayload } from "../../webhooks/payloads/order-confirmed-payload";
+import { SaleorOrderLine } from "../order-line";
+
+export interface ISaleorConfirmedOrderEvent {
+  getChannelSlug(): string;
+  getOrderId(): string;
+  isFulfilled(): boolean;
+  isStrategyFlatRates(): boolean;
+  getIsTaxIncluded(): boolean;
+  getLines(): SaleorOrderLine[];
+  getIsDiscounted(): boolean;
+  hasShipping(): boolean;
+  getShippingAmount(): number;
+}
```

Comment:
What is the point of this interface? 

1. Are you going to have different implementation of it?
2. Are you going to have some place where you expect interface instead of class instance?
3. ... and if yes, you will allow to inject something that pass, but also doesn't perform necessary validation

Interfaces work good for services, not for value objects/entities. In this case its antipattern. You want class with private constructor and factory method to validate data. Your interface break this concept

Reply:
You are right - it was leftover from previous iteration - removed